### PR TITLE
OSX safety

### PR DESCRIPTION
**Pull request by zacronos**
Made some changes to be more OS-agnostic in scripting

### CODE REVIEW

File: bin/compile [hunk 1]
```diff
@@ -7,17 +7,18 @@ main() {
     BUILD_DIR=$1    # /tmp/build_76471de2-7dff-46b7-b331-c8dd07653def
     CACHE_DIR=$2    # /app/tmp/cache
     ENV_DIR=$3
-    BUILDPACK_DIR=$(readlink -f $(dirname $(type -P "$0"))/..)
+    BUILDPACK_DIR="$( cd -P "$( dirname "${BASH_SOURCE[0]}" )" > /dev/null && pwd )/.."
```
Comment (zacronos) [inline bin/compile:10]: `readlink -f` isn't valid on OSX

File: bin/compile [hunk 2]
```diff
@@ -7,17 +7,18 @@ main() {
...
 
     load-env
 
-    NGINX_VERSION=${NGINX_VERSION-1.7.12}
-    NGINX_PCRE_VERSION=${NGINX_PCRE_VERSION-8.36}
-    NGINX_HEADERS_MORE_VERSION=${NGINX_HEADERS_MORE_VERSION-0.25}
+
+    NGINX_VERSION=${NGINX_VERSION-1.9.4}
+    NGINX_PCRE_VERSION=${NGINX_PCRE_VERSION-8.37}
+    NGINX_HEADERS_MORE_VERSION=${NGINX_HEADERS_MORE_VERSION-0.261}
```
Comment (zacronos) [inline bin/compile:17]: updated versions

File: bin/compile [hunk 3]
```diff
@@ -7,17 +7,18 @@ main() {
...
     
     
     declare cached_version
-    if ! is-enabled "$NGINX_FORCE_BUILD"; then
+    if [[ "$NGINX_FORCE_BUILD" != "true" ]]; then
```
Comment (zacronos) [inline bin/compile:21]: this is the only place `is-enabled` was being used; this is a little less general, but we're the only ones using this fork, so who cares

File: bin/compile [hunk 4]
```diff
@@ -55,6 +56,7 @@ build-nginx() {
     (
         set-indent
 
+        rm -rf $nginx_build_dir
```
Comment (zacronos) [inline bin/compile:59]: clean out its own build dir

File: bin/compile [hunk 5]
```diff
@@ -70,8 +72,7 @@ build-nginx() {
             --add-module=$PWD/headers-more-nginx-module-$NGINX_HEADERS_MORE_VERSION \
             --add-module=$PWD/nginx-delay-module \
             --with-http_gzip_static_module \
-            --with-http_realip_module \
-            --with-http_ssl_module
+            --with-http_realip_module
```
Comment (zacronos) [inline bin/compile:75]: turns out we don't need ssl on heroku, it handles that for us with an addon and we never see the ssl stuff

File: bin/compile [hunk 6]
```diff
@@ -99,17 +100,11 @@ install() {
 }
 
 set-indent() {
-    exec &> >(sed -u 's/^/       /')
+    exec &> >(sed 's/^/       /')
```
Comment (zacronos) [inline bin/compile:103]: `sed -u` isn't valid on OSX

File: bin/compile [hunk 7]
```diff
@@ -99,17 +100,11 @@ install() {
...
 
 announce() {
     echo "-----> nginx-buildpack: $*"
 }
 
-is-enabled() {
-    ( shopt -s extglob nocasematch
-      [[ $1 == @(1|true|yes|on) ]]
-    )
-}
```
Comment (zacronos) [inline bin/compile:113]: the `@()` syntax used here isn't valid on OSX

File: bin/start-nginx [hunk 8]
```diff
@@ -1,11 +1,19 @@
 #!/usr/bin/env bash
 
+set -e
```
Comment (zacronos) [inline bin/start-nginx:3]: for safety

File: bin/start-nginx [hunk 9]
```diff
@@ -1,11 +1,19 @@
...
+
+
 main() {
     declare psmgr=/tmp/nginx-buildpack-wait
-    declare -A pids signals
+    declare -a pids signals
```
Comment (zacronos) [inline bin/start-nginx:8]: `-A` isn't valid on OSX, only `-a`

File: bin/start-nginx [hunk 10]
```diff
@@ -1,11 +1,19 @@
...
 
     # Evaluate config to get environment variables
     mkdir -p config
+    if [[ "$OSTYPE" == "linux"* ]]; then
+        export NGINX_CONNECTION_METHOD=epoll
+    elif [[ "$OSTYPE" == "darwin"* || "$OSTYPE" == "freebsd"* ]]; then
+        export NGINX_CONNECTION_METHOD=kqueue
+    fi
```
Comment (zacronos) [inline bin/start-nginx:16]: do some OS detection and choose the best connection processing method available

**Comment by zacronos** [comment]
@nmccready @jessecstewart @realtyjustin 

This is a prerequisite before my other PR (to run nginx prod-style on dev) can be merged

**Comment by zacronos** [comment]
In fact, I'll wait until this is up to do a couple more tests, then I'll make the other PR

**Comment by zacronos** [comment]
Self-merging to keep things moving, if someone has objections I'll make another PR to address them

**Comment by jessecstewart** [comment]
LGTM

**Comment by nmccready** [comment]
Were specs important cause circle was failing?

**Comment by nmccready** [comment]
nvm there are no specs; be nice to correct circle to not report errors.

**Pull request merged**